### PR TITLE
Reduce amount of data going through the shuffle when writing single B…

### PR DESCRIPTION
**Pull request by tomwhite**
…AM file from Spark.

Currently we are sorting a key-value RDD where the key is a GATKRead and the value is the corresponding SAMRecord. This is very inefficient, since it doubles the amount of data going through the shuffle. It's much better to have a null key then convert to SAMRecord(Writable) just before writing to the file.

### CODE REVIEW

**Comment by tomwhite** [comment]
@davidadamsphd or @akiezun, can one of you review please?

File: src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java [hunk 1]
```diff
@@ -153,18 +153,11 @@ private static void writeReadsSingle(
...
-            SAMRecordWritable samRecordWritable = new SAMRecordWritable();
-            samRecordWritable.set(samRecord);
-            return new Tuple2<>(gatkRead, samRecordWritable);
-        });
+        JavaPairRDD<GATKRead, Void> rddReadPairs = rddReads.mapToPair(gatkRead -> new Tuple2<>(gatkRead, (Void) null));
 
         // do a total sort so that all the reads in partition i are less than those in partition i+1
-        final JavaPairRDD<GATKRead, SAMRecordWritable> out =
-                totalSortByKey(rddSamRecordWriteable, new ReadCoordinateComparator(header), GATKRead.class);
+        final JavaPairRDD<GATKRead, Void> out =
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java]: Could you add a comment here explaining why it's void?  I feel like someone is going to be confused in the future.

Comment (tomwhite) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java]: Done

**Comment by lbergelson** [comment]
looks good to me.  That's some pretty impressive performance increase you've accomplished.

**Pull request merged**